### DISCUSSION TITLE
Is it possible to check multiple properties on the same generated values?

### DISCUSSION THREAD

**Discussion by dan-blank** [dubzzz/fast-check#1539] [Q&A] [answered]
First, thank you and you all for this great library! I am very happy to see property-based testing in Typescript, and then in such a good state! :)

As for the question, consider this pseudo-ish code that I changed from the tutorial:

```
const reallyExpensiveToGenerateArb = <...>;

describe('properties', () => {

  it('Property X should hold', () => {
    fc.assert(fc.property(reallyExpensiveToGenerateArb , sth => doesXhold(sth)));
  });

  it('Property Y should hold', () => {
    fc.assert(fc.property(reallyExpensiveToGenerateArb , sth => doesYhold(sth)));
  });

});
```

So I want to have test different properties and have them 1) appear in mocha in the usual line-per-test fashion (properties -> Property X should hold ✔️ , properties -> Property Y should hold ❌ ) and 2) generate `reallyExpensiveToGenerateArb` only once per check against the properties. What I already did was to test all properties in one test - this way, we fulfill 2), but then if one check fails, the whole line is displayed as fail in mocha. I have also seen `fc.clone()` but it seems to me that it does not save computing work.

Is there a way to do what I want?

**Comment by dubzzz** [comment, marked as answer]
One of the solutions could be to use one expect per check:

```js​const reallyExpensiveToGenerateArb = <...>;
describe('properties', () => {
  it('Properties X, Y should hold', () => {
    fc.assert(fc.property(reallyExpensiveToGenerateArb , sth => {
        expect(doesXhold(sth)).toBe(true);
        expect(doesYhold(sth)).toBe(true);
   }));
  });
});
```

Would it works in your case?

**Reply by dan-blank** [reply]
It would generate the ´reallyExpensiveToGenerateArb´ only once which is good, but I would prefer splitting "Properties X, Y should hold" into two lines for the test framework, "Properties X should hold" and "Properties Y should hold". In a bigger project (with more properties), it would make a nicer looking test report, once could see immediately which properties are ✔️ and which are ❌.

But in the end, this is about aesthetics and not really important. If a test with 4 properties checked fails, then it is still obvious which property failed by the fast-check result. So it´s all perfectly usable. Thank you for your fast answer! :)
